cleanlab/object_detection/rank.py

```diff
@@ -0,0 +1,1040 @@
+            f"Pruning with threshold=={threshold} prunes {p_ann_pruned}% labels. Consider lowering the threshold.",
+            UserWarning,
+        )
+    if verbose:
+        print(f"Pruning {num_ann_to_zero} labels out of {total_ann} using threshold=={threshold}.")
+    return predictions_copy
+
+
+def _assert_valid_inputs(
+    labels: List[Dict[str, Any]],
+    predictions,
+    method: Optional[str] = None,
+    threshold: Optional[float] = None,
+):
+    """Asserts proper input format."""
+    if len(labels) != len(predictions):
+        raise ValueError(
+            f"labels and predictions length needs to match. len(labels) == {len(labels)} while len(predictions) == {len(predictions)}."
+        )
+    # Typecheck labels and predictions
+    if not isinstance(labels[0], dict):
+        raise ValueError(
+            f"Labels has to be a list of dicts. Instead it is list of {type(labels[0])}."
+        )
+    # check last column of predictions is probabilities ( < 1.)?
+    if not isinstance(predictions[0], (list, np.ndarray)):
+        raise ValueError(
+            f"Prediction has to be a list or np.ndarray. Instead it is type {type(predictions[0])}."
+        )
+    if not predictions[0][0].shape[1] == 5:
+        raise ValueError(
+            f"Prediction values have to be of format [x1,y1,x2,y2,pred_prob]. Please refer to the documentation for predicted probabilities under object_detection.rank.get_label_quality_scores for details"
+        )
+
+    valid_methods = ["objectlab"]
+    if method is not None and method not in valid_methods:
+        raise ValueError(
+            f"""
+            {method} is not a valid object detection scoring method!
+            Please choose a valid scoring_method: {valid_methods}
+            """
+        )
+
+    if threshold is not None and threshold > 1.0:
+        raise ValueError(
+            f"""
+            Threshold is a cutoff of predicted probabilities and therefore should be <= 1.
+            """
+        )
+
+
+def visualize(
+    image_path: str,
+    label: Dict[str, Any],
+    prediction: np.ndarray,
+    *,
+    prediction_threshold: Optional[float] = None,
+    given_label_overlay: bool = True,
+    class_labels: Optional[Dict[Any, Any]] = None,
+    figsize: Optional[Tuple[int, int]] = None,
+):
+    """Visualize bounding box labels (given labels) and model predictions for an image. The given labels
+    are shown with red while the predictions are shown in blue.
+
+    A single image is visualized with ``L`` annotated bounding boxes, ``M`` predicted bounding boxes and ``K`` total classes.
+
+
+    Parameters
+    ----------
+    image_path:
+        Full path to the image file.
+
+    label:
+        The given label for a single image in the format ``{'bboxes': np.ndarray((L,4)), 'labels': np.ndarray((L,))}`` where
+        ``L`` is the number of bounding boxes for the `i`-th image and ``bboxes[j]`` is in the format ``[x,y,x,y]`` with given label ``labels[j]``.
```

Comment:
Just say:

The given label for a single image, in the same format as each image's label for method_XYZ.

Here refer to the docstring of another method for description of labels. We don't want to give this highly technical description all over, just in select places. Same for prediction below